code/modules/mob/living/carbon/human/human_mob.dm

```diff
@@ -1686,49 +1686,152 @@ Eyes need to have significantly high darksight to shine unless the mob has the X
+				timer_restored = CPR_BREATHS_RESTORATION  // this, however, should keep it indefinitely postponed assuming CPR continues
+
+			if(HAS_TRAIT(H, TRAIT_FAKEDEATH))
+				if(prob(25))
+					to_chat(H, "<span class='userdanger'>Your chest burns as you receive unnecessary CPR!</span>")
+				continue
+
+			// this is the amount of time that should get added onto the timer.
+			var/new_time = (timer_restored + H.get_cpr_timer_adjustment(cpr_modifier))
+
+			SEND_SIGNAL(H, COMSIG_HUMAN_RECEIVE_CPR, (new_time SECONDS))
+
+			if(prob(5))
+				if(timer_restored > 4 SECONDS)
+					to_chat(pick(effective_cpr_messages))
+				else
+					to_chat(pick(ineffective_cpr_messages))
+
+			cpr_try_activate_bomb(H)
+
+
+		if(!H.IsRevivable())
+			to_chat(src, "<span class='notice'>You feel [H]'s body is already starting to stiffen beneath you...it's too late for CPR now.</span>")
+		else
+			visible_message("<span class='notice'>[src] stops giving [H] CPR.</span>", "<span class='notice'>You stop giving [H] CPR.</span>")
+
 		H.receiving_cpr = FALSE
-		to_chat(src, "<span class='danger'>You need to stay still while performing CPR!</span>")
+		return
+
+	visible_message("<span class='danger'>[src] is trying to perform CPR on [H.name]!</span>", "<span class='danger'>You try to perform CPR on [H.name]!</span>")
+	if(cpr_modifier == CPR_CHEST_COMPRESSION_ONLY)
+		to_chat(src, "<span class='warning'>You can't get to [H]'s mouth, so your CPR will be less effective!</span>")
+	while(do_mob(src, H, 4 SECONDS) && H.health <= HEALTH_THRESHOLD_CRIT)
+		H.adjustOxyLoss(-15 * cpr_modifier)
+		H.SetLoseBreath(0)
+		H.AdjustParalysis(-2 SECONDS)
+		H.updatehealth("cpr")
+		visible_message("<span class='danger'>[src] performs CPR on [H.name]!</span>", "<span class='notice'>You perform CPR on [H.name].</span>")
+
+		cpr_try_activate_bomb(H)
+
+		if(cpr_modifier == CPR_RESCUE_BREATHS)
+			to_chat(H, "<span class='notice'>You feel a breath of fresh air enter your lungs. It feels good.</span>")
+		add_attack_logs(src, H, "CPRed", ATKLOG_ALL)
+
+	H.receiving_cpr = FALSE
+	visible_message("<span class='notice'>[src] stops performing CPR on [H].</span>", "<span class='notice'>You stop performing CPR on [H].</span>")
+	to_chat(src, "<span class='danger'>You need to stay still while performing CPR!</span>")
 
 /mob/living/carbon/human/proc/get_cpr_mod(mob/living/carbon/human/H)
 	if(is_mouth_covered() || H.is_mouth_covered())
 		return CPR_CHEST_COMPRESSION_ONLY
 	if(!H.check_has_mouth() || !check_has_mouth())
 		return CPR_CHEST_COMPRESSION_ONLY
+	if(HAS_TRAIT(src, TRAIT_NOBREATH) || HAS_TRAIT(H, TRAIT_NOBREATH))
+		return CPR_CHEST_COMPRESSION_ONLY
+
+	if(isvox(src) && !isvox(H) || !isvox(src) && isvox(H) || isplasmaman(src) && !isplasmaman(H) || !isplasmaman(src) && isplasmaman(H))  // sorry non oxy-breathers
```

Comment:
```suggestion
	if(dna.species.breathid != H.dna.species.breathid)  // sorry non oxy-breathers
```

le simplify

Reply:
(please test this)

Reply:
I was looking all over for a variable that'd do this, thank you

Reply:
Tested, it works.